samcli/lib/sync/infra_sync_executor.py

```diff
@@ -174,8 +175,9 @@ def execute_infra_sync(self, first_sync: bool = False) -> InfraSyncResult:
             days_since_last_infra_sync = (current_time - last_infra_sync_time).days
 
         # Will not combine the comparisons in order to save operation cost
+        thread_id = uuid4()
         if self._sync_context.skip_deploy_sync and first_sync and (days_since_last_infra_sync <= AUTO_INFRA_SYNC_DAYS):
-            EventTracker.track_event("SyncFlowStart", "SkipInfraSyncExecute")
+            EventTracker.track_event("SyncFlowStart", "SkipInfraSyncExecute", thread_id=thread_id)
```

Comment:
I think you missed to add `thread_id` to line 191

Reply:
Yup I missed that, good catch